### ISSUE TITLE
Don't use colors in stdout/err if it's not a TTY

### ISSUE THREAD

**Issue by igsilya** [retis-org/retis#450]
Colors are nice, but they are hard to work with when using retis in scripts or CI systems.

```
Collector(s) started: skb-drop, skb-tracking, ovs, ct, skb
9 probe(s) loaded
ESC[0mESC[31mERROR ESC[0mCould not parse raw event: Factory 7 failed to parse section: Unsupported action id 24
ESC[0mESC[31mERROR ESC[0mCould not parse raw event: Factory 7 failed to parse section: Unsupported action id 24
ESC[0mESC[31mERROR ESC[0mCould not parse raw event: Factory 7 failed to parse section: Unsupported action id 24
ESC[0mESC[31mERROR ESC[0mCould not parse raw event: Factory 7 failed to parse section: Unsupported action id 24
ESC[0mESC[31mERROR ESC[0mCould not parse raw event: Factory 7 failed to parse section: Unsupported action id 24
ESC[0mESC[31mERROR ESC[0mCould not parse raw event: Factory 7 failed to parse section: Unsupported action id 24
ESC[0mESC[31mERROR ESC[0mCould not parse raw event: Factory 7 failed to parse section: Unsupported action id 24
Received signal, terminating...
```

**Comment by atenart** [comment]
When used in an automated way I would recommend using the `-o` option to write to a file and then use that (eg. `jq` or the new Python integration).

But you're right this should be better handled. We can detect if `TERM` is set (and not to `dumb`) to enable colors.

**Comment by igsilya** [comment]
> When used in an automated way I would recommend using the `-o` option to write to a file and then use that

Sure, but `-o` doesn't cover `stderr`, which I posted as an example.

**Comment by atenart** [comment]
Ah that's right as we split the output.

**Comment by igsilya** [comment]
Looks like https://doc.rust-lang.org/std/io/trait.IsTerminal.html is available since rust 1.70...

**Comment by atenart** [comment]
> Looks like https://doc.rust-lang.org/std/io/trait.IsTerminal.html is available since rust 1.70...

Nice!

I used a combination of this and `TERM`. We also have special cases like when using the internal pager where we do want to force color output. But the logic should cover most, if not all, cases. Let me know otherwise.
